utest/parsing/test_statements.py

```diff
@@ -0,0 +1,131 @@
+import unittest
+
+from robot.parsing import Token
+from robot.parsing.model.statements import (
+    Statement,
+    TestCaseName,
+    IfHeader,
+    ElseHeader,
+    ElseIfHeader,
+    End,
+    Comment,
+    EmptyLine
+)
+from robot.utils.asserts import assert_equal
+
+
+def assert_statement(model, expected):
+    assert_equal(model._fields, ('type', 'tokens'))
+    assert_equal(model.type, expected.type)
+    assert_equal(len(model.tokens), len(expected.tokens))
+    for m, e in zip(model.tokens, expected.tokens):
+        assert_equal(m, e, formatter=repr)
+    assert_equal(model._attributes, ('lineno', 'col_offset', 'end_lineno',
+                                     'end_col_offset', 'errors'))
+    assert_equal(model.lineno, expected.tokens[0].lineno)
+    assert_equal(model.col_offset, expected.tokens[0].col_offset)
+    assert_equal(model.end_lineno, expected.tokens[-1].lineno)
+    assert_equal(model.end_col_offset, expected.tokens[-1].end_col_offset)
+    assert_equal(model.errors, expected.errors)
```

Comment:
Should we perhaps add `__eq__` to `Statement`? Then comparing expected and actual statements would be trivial. `Token` already has it originally for same reasons. Implementation shouldn't require more than this:

```python
    def __eq__(self, other):
        return (isinstance(other, type(self))
                and self.tokens == other.tokens
                and self.errors == other.errors)

    def __ne__(self, other):
        return not self == other    
```

Better error reporting would then require adding also `__repr__`. I prototyped that a bit and this seems to produce pretty nice output:

```python
    def __repr__(self):
        errors = '' if not self.errors else ', errors=%s' % list(self.errors)
        return '%s(tokens=%s%s)' % (type(self).__name__, list(self.tokens), errors)
```

Alternatively we could use `ast.dump` in `__repr__` but I think repr that resembles how to construct the object is better.

Reply:
Good idea. We don't need any other comparison like `assert_equal(model.col_offset, expected.tokens[0].col_offset)` etc? I copied the whole comparison method from statements unit tests. But I guess if we're checking if we have the same tokens, it's enough as tokens are used to fill all other data.

Reply:
`Statement.col_offset` and others are properties. Their values depend solely on tokens so just testing tokens ought to be enough here. These properties themselves should be tested separately, and I think they already have some tests.

Reply:
Yes it's covered somewhere else. I used your code for __eq__